### PR TITLE
Add a bunch of unit tests!

### PR DESCRIPTION
**Pull request by bwateratmsft**
Fixes a bunch of #35. And fixes a couple of small bugs.

### CODE REVIEW

File: src/service/ComposeDocument.ts [hunk 1]
```diff
@@ -263,7 +263,7 @@ export class ComposeDocument {
                 // If the position is at the key separator, we're on the separator
                 pathParts.unshift(Sep);
                 pathParts.unshift(keyName);
-            } else if (params.position.character > indentLength) {
+            } else if (params.position.character >= indentLength) {
```
Comment (bwateratmsft) [inline src/service/ComposeDocument.ts:266]: This fixes an issue where the path was slightly incorrect, and hovering the mouse over the first character of a key did not correctly compute the path.

File: src/test/utils/debounce.test.ts [hunk 2]
```diff
@@ -3,10 +3,114 @@
  *  Licensed under the MIT License. See LICENSE in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
+import { debounce } from '../../service/utils/debounce';
+
+// This delay is used as the delay given to `debounce`, but also assertions are made that each test should take at least (delay - 1) milliseconds and at most 2 * delay milliseconds
+// This could result in some test fragility; if so a longer delay should reduce the frequency of that
```
Comment (karolz-ms) [inline src/test/utils/debounce.test.ts:9]: The lower bound is OK, but the upper bound, in my experience, is just an invitation to have regular, annoying false positive failures. The cloud VMs running tests are prone to very long pauses (10s of seconds) at random times.

If you really want to ensure that the upper bound is holding in a reasonable way, run the test 100 times or so, drop the 10 lowest and 10 highest results and then check the median of the remaining results.

**Review by karolz-ms — APPROVED** [review]
very nice

File: src/test/utils/debounce.test.ts (hunk 2 above)
Comment (bwateratmsft) [reply src/test/utils/debounce.test.ts:9]: Good point. I added an assertion failure in the first test to be extra emphatic that _only_ the second debounce call should get executed (the value assertion for `x` only ensures that one of them gets executed, not particularly which one). I removed the upper bound on time.

**Pull request merged**